blocks/blogsearch/blogsearch.js

```diff
@@ -1,15 +1,69 @@
 import {
-  div, form, input, i,
+  a, div, form, input, i,
 } from '../../scripts/dom-helpers.js';
+import { getLocaleInfo, debounce } from '../../scripts/scripts.js';
+import ffetch from '../../scripts/ffetch.js';
+
+async function handleSearch(block) {
+  const searchValue = block.querySelector('input').value;
+  if (searchValue.length < 3) {
+    return;
+  }
+
+  const searchResults = block.querySelector('.search-results');
+  searchResults.innerHTML = '';
+
+  console.log(`searching for ${searchValue}`);
+
+  const entries = await ffetch(`${getLocaleInfo().metadataIndex}`)
```

Comment:
Even if the request gets cached, I think we shouldn't do a fetch every time there is a search, but rather save the content of the sheet in memory and then work directly on the data from memory. See how I did it in https://github.com/hlxsites/servicenow/blob/main/scripts/scripts.js#L142 . 

So we can essentially save the content of the `blogs-content` on the first search, then simply apply new filters on it for every change of the input.

Reply:
I initially asked myself the same thing, but I found it done in `aem.js`, so this makes me think that there should be some benefit to it. :) https://github.com/hlxsites/servicenow/blob/main/scripts/aem.js#L471

Additionally, we would be doing an initial filtering based on locales, so the number of entries to iterate over in the subsequent searches will always be smaller than getting all of them again.

So at least this kind of filtering: https://github.com/hlxsites/servicenow/pull/71/files#diff-fbceb58f32b99b8a5309251f8c49ac8eda82fba58b905c600c92bce1ed2a1c8cR159 will happen only once.

Reply:
thanks for explaining -- on it 👍

Reply:
just a side note: saving the result loses us the ability to use ffetch's flexibility of generator functions up to limiting the result set (and stopping from requesting chunks). 
but I assume that's rather academical with the number of blog articles we have we'll probably never need that so no point compromising for a scenario that we don't anticipate

Reply:
done in 7d07f47